server/services/v1/database/query_runner.go

```diff
@@ -755,3 +757,65 @@ func (runner *StreamingQueryRunner) writeTimestamp(buf *bytes.Buffer, key string
 
 	return nil
 }
+
+type ExplainQueryRunner struct {
+	*BaseQueryRunner
+
+	req *api.ReadRequest
+}
+
+func (runner *ExplainQueryRunner) Run(ctx context.Context, tx transaction.Tx, tenant *metadata.Tenant) (Response, context.Context, error) {
+	db, err := runner.getDatabase(ctx, nil, tenant, runner.req.GetProject(), runner.req.GetBranch())
+	if err != nil {
+		return Response{}, ctx, err
+	}
+
+	ctx = runner.cdcMgr.WrapContext(ctx, db.Name())
+
+	collection, err := runner.getCollection(db, runner.req.GetCollection())
+	if err != nil {
+		return Response{}, ctx, err
+	}
+
+	options, err := runner.buildReaderOptions(runner.req, collection)
+	if err != nil {
+		return Response{}, ctx, err
+	}
+
+	explainResp, err := buildExplainResp(options, collection, runner.req.Filter)
+	if err != nil {
+		return Response{}, ctx, err
+	}
+
+	return Response{
+		Response: explainResp,
+	}, ctx, nil
+}
+
+const (
+	SEARCH    = "search"
+	PRIMARY   = "scan"
+	SECONDARY = "secondary index"
+)
+
+func buildExplainResp(options readerOptions, coll *schema.DefaultCollection, filter []byte) (*api.ExplainResponse, error) {
```

Comment:
Don't we need to have one for SCAN as well, i.e., when a full table scan is triggered?

Reply:
That is on line 819.